docs/introduction/why-rtk-is-redux-today.md

```diff
@@ -0,0 +1,220 @@
+- `applyMiddleware` to combine multiple middleware into a store enhancer
+- `compose` to combine multiple store enhancers into a single store enhancer
+
+Other than that, all the other Redux-related logic in your app has to be written entirely by you.
+
+The good news is that this means Redux _can_ be used in many different ways. The bad news is that there are no helpers to make any of your code easier to write.
+
+For example, a reducer function is _just_ a function. Prior to Redux Toolkit, you'd typically write it that reducer with a `switch` statement and manual updates. You'd also probably have hand-written action creators and action type constants along with it:
+
+```js title="Legacy hand-written Redux usage"
+const ADD_TODO = 'ADD_TODO'
+const TODO_TOGGLED = 'TODO_TOGGLED'
+
+export const addTodo = text => ({
+  type: ADD_TODO,
+  payload: { text, id: nanoid() }
+})
+
+export const todoToggled = id => ({
+  type: ADD_TODO,
+  payload: id
+})
+
+export const todosReducer = (state = [], action) => {
+  switch (action.type) {
+    case ADD_TODO:
+      return state.concat({
+        id: action.payload.id,
+        text: action.payload.text,
+        completed: false
+      })
+    case TODO_TOGGLED:
+      return state.map(todo => {
+        if (todo.id !== action.payload) return
+
+        return {
+          ...todo,
+          completed: !todo.completed
+        }
+      })
+    default:
+      return state
+  }
+}
+```
+
+None of this code specifically depends on any API from the `redux` core library. But, this is a lot of code to write. Immutable updates required a lot of hand-written object spreads and array operations, and it was very easy to make mistakes and accidentally mutate state in the process. It was also common, though not strictly required, to spread that same code across multiple files: `actions/todos.js`, `constants/todos.js`, and `reducers/todos.js`.
+
+Additionally, store setup usually required a few dozen lines of setting up commonly used middleware like thunks, hooking up the Redux DevTools Extension support
+
+### What Does Redux Toolkit Do?
+
+While these _were_ the patterns originally shown in the Redux docs, they unfortunately require a lot of very verbose and repetitive code. Most of this boilerplate isn't _necessary_ to use Redux. On top of that, the boilerplate-y code lead to more opportunities to make mistakes.
+
+**We specifically created Redux Toolkit to eliminate the "boilerplate" from hand-written Redux logic, prevent common mistakes, and provide APIs that simplify standard Redux tasks**.
+
+Redux Toolkit starts with two key APIs that simplify the most common things you do in every Redux app:
+
+- `configureStore` sets up a well-configured Redux store with a single function call, including combining reducers, adding the thunk middleware, and setting up the Redux DevTools integration. It also is easier to configure than `createStore`, because it takes named options parameters.
```

Comment:
```suggestion
- `configureStore` sets up a well-configured Redux store with a single function call, including combining reducers, adding the thunk middleware (you can opt out of it if you don't want it), and setting up the Redux DevTools integration. It also is easier to configure than `createStore`, because it takes named options parameters.
```